src/hooks/useCart.ts

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect, useCallback } from "react";
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "../modules/Store";
+import { getCarts, deleteCart, totalCheck } from "../modules/Carts";
+import { postOrder } from "../modules/Orders";
+
+function useCart() {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const [totalPrice, setTotalPrice] = useState(0);
+  const [totalChecked, setTotalChecked] = useState(true);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+
+  const carts = useSelector((state: RootState) => state.carts.cartsList);
+
+  const checkedCartList = carts
+    .filter((cart) => cart.checked === true)
+    .map((cart) => {
+      return { ...cart.product, quantity: cart.quantity };
+    });
+
+  const handleTotalChecked = () => {
+    setTotalChecked(!totalChecked);
+  };
+
+  const handleTotal = useCallback(() => {
+    setTotalPrice(
+      carts
+        .map((cart) => {
+          if (cart.checked) {
+            return cart.product.price * cart.quantity;
+          }
+          return 0;
+        })
+        .reduce((previous, current) => previous + current, 0)
```

Comment:
```suggestion
      carts
        .map((cart) => cart.checked ? cart.product.price * cart.quantity : 0)
        .reduce((previous, current) => previous + current, 0)
```

저는 이런 식으로 return 문에 삼항 연산자를 활용하는걸 좋아하는데 진수님은 대체로 삼항연산자를 지양하시는 편일까요?

Reply:
다른 부분들에서도 대체로 삼항연산자를 사용하지 않으시는거 같아서 궁금해서 여쭤밨습니다

Reply:
저는 삼항 연산자를 지양하는 편은 아니고 기능 추가를 염두에 둔다면 return 문을 사용하여 각각 피처로 만들고, 그렇지 않고 이번 케이스처럼 피처의 확장성을 생각하지 않아도 되는 단순 연산이나 렌더링 같은 경우는 삼항 연산자를 사용합니다~
말씀해주신 대로 이번 케이스에서는 삼항 연산자가 더욱 어울리?리고 제가 사용하는 기준에서 맞습니다!!